rclcpp/include/rclcpp/executors/multi_threaded_executor.hpp

```diff
@@ -81,12 +82,17 @@ class MultiThreadedExecutor : public rclcpp::Executor
 private:
   RCLCPP_DISABLE_COPY(MultiThreadedExecutor)
 
-  std::mutex wait_mutex_;
+  std::mutex wait_mutex_;  // Unused. Leave it for ABI compatibility.
   size_t number_of_threads_;
   bool yield_before_execute_;
   std::chrono::nanoseconds next_exec_timeout_;
 
   std::set<TimerBase::SharedPtr> scheduled_timers_;
+  static std::unordered_map<MultiThreadedExecutor *,
+    std::shared_ptr<detail::MutexTwoPriorities>> wait_mutex_set_;
+  static std::mutex shared_wait_mutex_;
+  // These variables are declared as static variables for ABI-compatibiliity.
+  // And they mimic member variables needed to backport from master.
```

Comment:
i am not very much positive to have this giant lock in the process space to keep the compatibility. (but i do not have good solution either, sorry) if i am not mistaken, we would be able to keep the ABI for Linux with this fix, but what about windows and OSX?

Reply:
Globals don't affect ABI compatibility in any of them AFAIK

Reply:
okay, thanks for the info.

Reply:
How about declaring them as static global variables in multi_threaded_executor.cpp and make them file scope?

Reply:
@fujitatomoya 
I'm a little confused about the implementation, if you don't mind, could you give me your opinion about this?
>　How about declaring them as static global variables in multi_threaded_executor.cpp and make them file scope?

Reply:
sorry to be late to get back to you, thanks for the ping.
after all, i think current implementation is okay, for not breaking ABI, we eventually need to have these in global area.

Reply:
Thanks for the comment.
I've changed this Draft to PR for now.